### PR TITLE
Fix/precip  to address inconsistent treatment of precipitation 

### PR DESCRIPTION
**Pull request by rmcd1024**
This modifies the precipitation calculation in weather.r to fix an inconsistency. In the raw weather data, precipitation totals are cumulative for the hour and reset to 0 at 51 minutes. Sometimes there is only one weather record for an hour, at 51 minutes, in which case p01i  is the correct hourly total. Other times there are multiple records within an hour, in which case it is necessary to find the maximum reported precipitation earlier than or equal to 51 minutes. (And there are times when there is no 51 minute record.) The existing weather.r takes the first record in the hour, which sometimes gives the hourly cumulative rainfall, but at other times gives whatever rainfall is reported in the first record that hour.

For example consider LGA 2013-06-08 01:00 to 02:00. There are 5 records in the raw data. The first record at 1:16 reports 0.18", which is the precip value reported in the weather data in the package. But the actual hourly cumulative rainfall is for that hour was 0.24" (reported at 01:49 --- this is a case where there is no 51 minute record).

I have been unable to find documentation for the 51 minutes reset time, but it is evident in the data. The code in the pull request generates hourly precipitation that when cumulated matches the official national weather service daily totals (sometimes there is a .01" difference which is made up the next day. I'm not sure why that happens). 

For more details there is [a stackexchange discussion](https://earthscience.stackexchange.com/questions/10598/how-are-daily-precipitation-totals-computed) that I initiated about this issue.

Note that because the raw data download format has changed, the existing weather.r doesn't work, so this fix cannot be tested without also implementing my pull request #24 . I wasn't sure how to handle this so I kept them separate.

### CODE REVIEW

**Comment by rmcd1024** [comment]
Daryl Herzmann at Iowa State was kind enough to respond to my request for more information. He has [a presentation on rainfall measurement](https://mesonet.agron.iastate.edu/present/130903_isu/isumet_fall2013_web.pdf). He tells me that hourly resets of rainfall totals can occur between 51 and 57 minutes, depending on the site, and that daily rainfall totals are measured midnight to midnight, local standard time.

**Comment by hadley** [comment]
Thanks!

Do you mind briefly summarising the discussion in a comment?

(And adding a news bullet)

File: data-raw/weather.R [hunk 1]
```diff
@@ -59,8 +59,11 @@ weather <- raw %>%
     wind_speed = as.numeric(wind_speed) * 1.15078, # convert to mpg
     wind_gust = as.numeric(wind_speed) * 1.15078
   ) %>%
-  mutate(year = 2013, month = month(time), day = mday(time), hour = hour(time)) %>%
+  mutate(year = 2013, month = month(time), day = mday(time),
+         hour = hour(time), minute = minute(time)) %>%
   group_by(station, month, day, hour) %>%
+  filter(minute <= 51) %>%
```
Comment (hadley) [inline data-raw/weather.R]: Could you please insert a comment here briefly explaining (with link) why we filter this out, and use `max()` as a summary?

File: data-raw/weather.R [hunk 2]
```diff
@@ -59,8 +59,11 @@ weather <- raw %>%
...
+  mutate(precip = max(precip)) %>%
   filter(row_number() == 1) %>%
```
Comment (hadley) [inline data-raw/weather.R]: Do we still need this line or is that a different problem?

**Comment by rmcd1024** [comment]
There are cases where there are multiple records prior to minute 51. So
it's necessary to extract a single record. I'm not sure it matters how it's
done once precip is defined in the preceding mutate.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**